ghcide/src/Development/IDE/Plugin/TypeLenses.hs

```diff
@@ -90,15 +87,24 @@ descriptor plId =
     { pluginHandlers = mkPluginHandler STextDocumentCodeLens codeLensProvider
     , pluginCommands = [PluginCommand (CommandId typeLensCommandId) "adds a signature" commandHandler]
     , pluginRules = rules
+    , pluginCustomConfig = mkCustomConfig properties
     }
 
+properties :: Properties '[PropertyKey "mode" 'TEnum]
+properties = emptyProperties
+  & defineEnumProperty @"mode" "Control how type lenses are shown"
```

Comment:
Using `-XOverloadedLabels` might make this type-application cleaner.

Reply:
I haven't used this extension before.. will take a look

Reply:
Essentially:

```haskell

data Key (s :: Symbol) where
  Key :: KnownSymbol s => Key s

instance (s ~ s', KnownSymbol s) => IsLabel s (Key s') where
  fromLabel = Key
```

Now `defineEnumProperty` can have type `Key -> ...`, and you can change this call site to be 

```haskell
defineEnumProperty #mode "Control how type lenses are shown"
```

Reply:
I see, thiank you!